pkgs/by-name/ho/homebox/package.nix

```
@@ -0,0 +1,145 @@
+{
+  stdenvNoCC,
+  jq,
+  moreutils,
+  nodePackages,
+  cacert,
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}: let
+  pname = "homebox";
+  version = "0.10.3";
+  src = fetchFromGitHub {
+    owner = "hay-kot";
+    repo = "homebox";
+    rev = "v${version}";
+    hash = "sha256-Hej/dM0BgtRWiMOpp/SDVr3H1IbYb935T1pfX8apjpE=";
+    # Inspired by: https://github.com/NixOS/nixpkgs/blob/f2d7a289c5a5ece8521dd082b81ac7e4a57c2c5c/pkgs/applications/graphics/pdfcpu/default.nix#L20-L32
+    # The intention here is to write the information into files in the `src`'s
+    # `$out`, and use them later in other phases (in this case `preBuild`).
+    # In order to keep determinism, we also delete the `.git` directory
+    # afterwards, imitating the default behavior of `leaveDotGit = false`.
+    # More info about git log format can be found at `git-log(1)` manpage.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git log -1 --pretty=%H > "backend/COMMIT"
+      git log -1 --pretty=%cd --date=format:'%Y-%m-%dT%H:%M:%SZ' > "backend/SOURCE_DATE"
+      rm -rf ".git"
+    '';
+  };
+  unsupported = throw "Unsupported system ${stdenvNoCC.hostPlatform.system} for ${pname}";
+
+  frontend = stdenvNoCC.mkDerivation {
+    pname = "${pname}-frontend";
+    inherit version;
+
+    src = "${src}/frontend";
+
+    preBuild = ''
+      export HOME=$(mktemp -d)
+      export STORE_PATH=$(mktemp -d)
+
+      pnpm config set store-dir "${pnpm-deps}"
+      pnpm install --offline --frozen-lockfile --shamefully-hoist
+      patchShebangs node_modules/{*,.*}
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+
+      pnpm build
+
+      runHook postBuild
+    '';
+    env.NUXT_TELEMETRY_DISABLED = 1;
+
+    nativeBuildInputs = [
+      nodePackages.pnpm
```

```suggestion
      pnpm_9
```
please use the top-level pnpm